Common/Orders/TimeInForce.cs

```diff
@@ -0,0 +1,159 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using QuantConnect.Interfaces;
+using QuantConnect.Orders.TimeInForces;
+using QuantConnect.Securities;
+
+namespace QuantConnect.Orders
+{
+    /// <summary>
+    /// Time In Force - defines the length of time over which an order will continue working before it is canceled
+    /// </summary>
+    public abstract class TimeInForce : IEquatable<TimeInForce>, ITimeInForceHandler
+    {
+        /// <summary>
+        /// Gets a <see cref="TimeInForce"/> instance with <see cref="TimeInForceType.GoodTilCanceled"/> type
+        /// </summary>
+        public static readonly TimeInForce GoodTilCanceled = new GoodTilCanceledTimeInForce();
+
+        /// <summary>
+        /// Gets a <see cref="TimeInForce"/> instance with <see cref="TimeInForceType.Day"/> type
+        /// </summary>
+        public static readonly TimeInForce Day = new DayTimeInForce();
+
+        /// <summary>
+        /// The type of Time In Force
+        /// </summary>
+        public TimeInForceType Type { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TimeInForce"/> class
+        /// </summary>
+        protected TimeInForce(TimeInForceType type)
+        {
+            Type = type;
+        }
+
+        /// <summary>
+        /// Converts a <see cref="TimeInForceType"/> value to a new <see cref="TimeInForce"/> instance.
+        /// </summary>
+        public static implicit operator TimeInForce(TimeInForceType type)
+        {
+            switch (type)
+            {
+                case TimeInForceType.GoodTilCanceled:
+                    return new GoodTilCanceledTimeInForce();
+
+                case TimeInForceType.Day:
+                    return new DayTimeInForce();
+
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(type), type, $"Implicit conversion from TimeInForceType.{type} to TimeInForce not supported.");
+            }
+        }
+
+        /// <summary>
+        /// Returns a <see cref="TimeInForceType"/> value from a <see cref="TimeInForce"/> instance.
+        /// </summary>
+        public static implicit operator TimeInForceType(TimeInForce timeInForce)
+        {
+            return timeInForce.Type;
+        }
+
+        /// <summary>
+        /// Equals operator
+        /// </summary>
+        /// <param name="left">The left operand</param>
+        /// <param name="right">The right operand</param>
+        /// <returns>True if both symbols are equal, otherwise false</returns>
+        public static bool operator ==(TimeInForce left, TimeInForce right)
+        {
+            if (ReferenceEquals(left, right)) return true;
+            if (ReferenceEquals(left, null)) return false;
+            if (ReferenceEquals(right, null)) return false;
+
+            return left.Type == right.Type;
```

Comment:
This isn't 100% correct. Consider two GTD implementations with different end dates (or different time spans if we make that change). Best bet is to `return left.Equals(right)` and let the implementations decide how to resolve equality. Similar comment for all equality methods in this class.

Reply:
The idea here was to have a common default implementation of equality methods (to avoid duplicating these methods in derived classes with no additional properties -- such as GTC and DAY) , but since I'm going to remove the `TimeInForceType Type` property, this will be changed to `left.Equals(right)`.